src/realm/table.cpp

```diff
@@ -2157,6 +2157,42 @@ void Table::insert_empty_row(size_t row_ndx, size_t num_rows)
     }
 }
 
+size_t Table::add_row_with_key(size_t key_col_ndx, int64_t key)
+{
+    size_t num_cols = m_spec.get_column_count();
+    size_t row_ndx = m_size;
+
+    REALM_ASSERT(is_attached());
+    REALM_ASSERT_3(key_col_ndx, <, num_cols);
+
+    if (REALM_UNLIKELY(num_cols == 0)) {
+        throw LogicError(LogicError::table_has_no_columns);
+    }
+
+    bump_version();
+
+    for (size_t col_ndx = 0; col_ndx != num_cols; ++col_ndx) {
+        if (col_ndx == key_col_ndx) {
+            auto& col = get_column(key_col_ndx);
```

Comment:
I would prefer `IntegerColumn&` for clarity, but perhaps I should just get used to `auto`. `get_column` is probably the "true" source of the readability problem here.